engine/sound/src/sound.cpp

```diff
@@ -382,6 +432,8 @@ namespace dmSound
         sound->m_HasWindowFocus = true; // Assume we startup with the window focused
         sound->m_DeviceType = device_type;
         sound->m_Device = device;
+        sound->m_DeviceParams = device_params; // Stash frame and buffer count for potential device reset
+        sound->m_DeviceResetPending = false;
```

Comment:
would it make sense to call ResetDevice() in the Initialze() function?
(To avoid duplicated code for setting up a device)